scripts/mbedtls_framework/outcome_analysis.py

```diff
@@ -112,11 +114,19 @@ def name_matches_pattern(name: str, str_or_re: IgnoreEntry) -> bool:
     else:
         return str_or_re == name
 
+def open_outcome_file(outcome_file: str) -> typing.TextIO:
+    if outcome_file.endswith('.gz'):
+        return gzip.open(outcome_file, 'rt', encoding='utf-8')
+    elif outcome_file.endswith('.xz'):
+        return lzma.open(outcome_file, 'rt', encoding='utf-8')
+    else:
+        return open(outcome_file, 'r', encoding='utf-8')
```

Comment:
Question: why are we using `rt` for gzip/lzma open and just `r` here? (Not saying that's wrong, just curious.)

Reply:
Because `t` is the default for the `open` builtin, but not for gzip and its imitators. And the existing code had `'r'` and I hadn't bothered to change it.

I'd actually considered writing this as
```
if …:
   open_function = …
else:
    open_function = open
open(outcome_file, 'r', encoding='utf-8')
```
but felt it was a bit too Haskelly for a Python script.

Reply:
That's probably how I would have written it, so I should probably check out Haskell next time I want to learn a new language ;)

Reply:
Using an `open_function` variable doesn't work because the candidate functions have different types and (our current version of) mypy isn't smart enough to reconcile them without help. This isn't worth spending time figuring out if it can be helped.